.github/workflows/ci.yml

```diff
@@ -21,29 +21,59 @@ jobs:
         include:
         - os: ubuntu-latest
           cibw_build: "cp36-*"
+          cibw_archs_linux: "auto64"
+          cibw_archs_macos: "auto64"

```

Comment:
If anyone knows how to avoid this ridiculous level of duplication, that would be much appreciated.

Reply:
I *think* you could define these under `matrix` and then override the default for the aarch64 element. Depending on how much work you want to do, you may also be able to combine all this logic in the environment variables for cibuildwheel: https://github.com/vispy/vispy/blob/182f87d62baa7158db96d963c7da435b723178cc/.github/workflows/wheels.yml#L22-L35

Reply:
> I think you don't need to set `cibw_archs_macos` matrix variable when running on ubuntu, nor `cibw_archs_linux` or `cibw_archs_macos` when running on windows, etc. They will default to empty variable and should be ignored anyway because they are not used by cibuildwheel unless on the target platform.

See here what happened when I did that: https://github.com/rougier/freetype-py/runs/5008175186?check_suite_focus=true#step:4:45

`ValueError: '' is not a valid Architecture`

Reply:
> I _think_ you could define these under `matrix` and then override the default for the aarch64 element.

Yeah so _how exactly_ that can be done is unclear to me.

> Depending on how much work you want to do, you may also be able to combine all this logic in the environment variables for cibuildwheel: https://github.com/vispy/vispy/blob/182f87d62baa7158db96d963c7da435b723178cc/.github/workflows/wheels.yml#L22-L35

As you can see from the currently running build, the aarch64 jobs independently take _very_ long because they run via QEMU. Using one job per OS would go over global build timeouts, and be a pretty terrible developer experience, I'm pretty sure.

Reply:
> Using one job per OS would go over global build timeouts

Isn't that what's already happening? One matrix "entry" for each operating system/arch? I'm also not sure I understand the `cibw_build` in this matrix. Some of these look like they are only building `cp36-*` wheels? Or only 3.8?

I might not know what I'm talking about. It's been a while since I've tried to get cibuildwheel to do what I want.